controllers/cloud.redhat.com/clowdapp_controller.go

```diff
@@ -138,192 +133,24 @@ func (r *ClowdAppReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
-		return ctrl.Result{}, nil
-	}
-
-	ans := &core.Namespace{}
-	if getAppNSErr := r.Client.Get(ctx, types.NamespacedName{Name: app.Namespace}, ans); getAppNSErr != nil {
-		return ctrl.Result{Requeue: true}, getAppNSErr
-	}
-
-	if ans.ObjectMeta.DeletionTimestamp != nil {
-		log.Info("App namespace is to be deleted - skipping reconcile")
-		return ctrl.Result{}, nil
-	}
-
-	if env.Generation != env.Status.Generation {
-		r.Recorder.Eventf(&app, "Warning", "ClowdEnvNotReconciled", "Clowder Environment [%s] is not reconciled", app.Spec.EnvName)
-		log.Info("Env not yet reconciled")
-		if setClowdStatusErr := SetClowdAppConditions(ctx, r.Client, &app, crd.ReconciliationFailed, fmt.Errorf("clowd env not reconciled")); setClowdStatusErr != nil {
-			log.Info("Set status error", "err", setClowdStatusErr)
-			return ctrl.Result{Requeue: true}, setClowdStatusErr
-		}
-		return ctrl.Result{}, nil
-	}
-
-	if !env.IsReady() {
-		r.Recorder.Eventf(&app, "Warning", "ClowdEnvNotReady", "Clowder Environment [%s] is not ready", app.Spec.EnvName)
-		log.Info("Env not yet ready")
-		if setClowdStatusErr := SetClowdAppConditions(ctx, r.Client, &app, crd.ReconciliationFailed, fmt.Errorf("clowd env not ready")); setClowdStatusErr != nil {
-			log.Info("Set status error", "err", setClowdStatusErr)
-			return ctrl.Result{Requeue: true}, setClowdStatusErr
-		}
-
-		return ctrl.Result{Requeue: true}, nil
-	}
-
 	cacheConfig := rc.NewCacheConfig(Scheme, errors.ClowdKey("log"), ProtectedGVKs, DebugOptions)
-
 	cache := rc.NewObjectCache(ctx, r.Client, cacheConfig)
-	//cache := providers.NewObjectCache(ctx, r.Client, scheme)
-
-	provider := providers.Provider{
-		Client: r.Client,
-		Ctx:    ctx,
-		Env:    &env,
-		Cache:  &cache,
-		Log:    log,
-	}
 
-	if provErr := r.runProviders(log, &provider, &app); provErr != nil {
-		r.Recorder.Eventf(&app, "Warning", "FailedReconciliation", "Clowdapp requeued [%s]", app.GetClowdName())
-		if setClowdStatusErr := SetClowdAppConditions(ctx, r.Client, &app, crd.ReconciliationFailed, provErr); setClowdStatusErr != nil {
-			log.Info("Set status error", "err", setClowdStatusErr)
-			return ctrl.Result{Requeue: true}, setClowdStatusErr
-		}
-		log.Info("Provider error", "err", provErr)
-		return ctrl.Result{Requeue: true}, provErr
-	}
-
-	cacheErr := cache.ApplyAll()
+	log.Info("Reconciliation started")
 
-	if cacheErr != nil {
-		r.Recorder.Eventf(&app, "Warning", "FailedReconciliation", "Clowdapp requeued [%s]", app.GetClowdName())
-		if setClowdStatusErr := SetClowdAppConditions(ctx, r.Client, &app, crd.ReconciliationFailed, cacheErr); setClowdStatusErr != nil {
-			log.Info("Set status error", "err", setClowdStatusErr)
-			return ctrl.Result{Requeue: true}, setClowdStatusErr
+	reconciliation := MakeClowdAppReconciliation(ctx, &cache, r.Client, r.Recorder, &app, log, &req)
+	res, err := reconciliation.Reconcile()
+	if err != nil {
+		if shouldSkipReconciliation(err) {
+			return ctrl.Result{}, nil
```

Comment:
Ah ok, it should be 

```
	res, err := reconciliation.Reconcile()
	if err != nil {
		if shouldSkipReconciliation(err) {
			return res, nil
		}
		return res, err
	}
```

Because some skips requeue:
```go
func (r *ClowdAppReconciliation) isEnvLocked() (ctrl.Result, error) {
	if ReadEnv() == r.app.Spec.EnvName {
		r.recorder.Eventf(r.app, "Warning", "ClowdEnvLocked", "Clowder Environment [%s] is locked", r.app.Spec.EnvName)
		r.log.Info("Env currently being reconciled")
		return ctrl.Result{Requeue: true}, errors.New(SKIPRECONCILE)
	}
	return ctrl.Result{}, nil
}
```

And some don't:
```go
func (r *ClowdAppReconciliation) isAppDisabled() (ctrl.Result, error) {
	if r.app.Spec.Disabled {
		r.log.Info("Reconciliation aborted - set to be disabled")
		return ctrl.Result{}, errors.New(SKIPRECONCILE)
	}
	return ctrl.Result{}, nil
}
```

Which matches the behavior of the old code:
```go

	if ReadEnv() == app.Spec.EnvName {
		r.Recorder.Eventf(&app, "Warning", "ClowdEnvLocked", "Clowder Environment [%s] is locked", app.Spec.EnvName)
		log.Info("Env currently being reconciled")
		return ctrl.Result{Requeue: true}, nil
	}
...
	if app.Spec.Disabled {
		log.Info("Reconciliation aborted - set to be disabled")
		return ctrl.Result{}, nil
	}
```

I changed it to reture the result - the steps were already doing the right thing but the skip eval wasn't respecting requeues